x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -328,6 +330,14 @@ protected PhysicalPlan rule(AggregateExec aggregateExec) {
             if (aggregateExec.child() instanceof EsQueryExec queryExec) {
                 var tuple = pushableStats(aggregateExec);
 
+                // for the moment support pushing count just for one field
+                List<Stat> stats = tuple.v2();
+                if (stats.size() > 1) {
```

Comment:
Yes since the rule might pick up two different fields however at runtime we don't know how to combine the operators.
For example `count(salary), count(emp_no) by gender` - this results into one EsStatsQuery with two Stat however because these are two different queries, two different sources are required and we currently allow only one.